relatedartist.go

```diff
@@ -24,29 +24,66 @@ func NewRelatedArtistService(auth *auth.Authenticator) *RelatedArtistService {
 	}
 }
 
-func (ras *RelatedArtistService) getByArtistId(id string) []relatedartist {
+func (ras *RelatedArtistService) getTopTracksFromRelatedArtists(id string, depth int) ([]spotify.FullTrack, error) {
 	client := ras.Auth.Client
 
+	topTracks, err := client.GetArtistsTopTracks(spotify.ID(id), "US")
+	if err != nil {
+		log.Fatalf("error retrieving top tracks data: %v", err)
+		return nil, err
+	}
+
 	relatedArtists, err := client.GetRelatedArtists(spotify.ID(id))
 	if err != nil {
 		log.Fatalf("error retrieving related artist data: %v", err)
+		return nil, err
 	}
 
-	artists := make([]relatedartist, len(relatedArtists))
-
-	for i, a := range relatedArtists {
-		topTracks, err := client.GetArtistsTopTracks(a.ID, "US")
+	for _, artist := range relatedArtists {
+		artistTopTracks, err := client.GetArtistsTopTracks(artist.ID, "US")
 		if err != nil {
-			log.Fatalf("error retrieving top tracks data: %v", err)
+			log.Fatalf("error retrieving top tracks data for related artist %s: %v", artist.Name, err)
+			return nil, err
 		}
+		topTracks = append(topTracks, artistTopTracks...)
 
-		artists[i] = relatedartist{
-			name:       a.Name,
-			id:         string(a.ID),
-			popularity: a.Popularity,
-			topTracks:  topTracks,
+		if depth > 1 {
+			relatedArtistTopTracks, err := ras.getTopTracksFromRelatedArtists(artist.ID.String(), depth-1)
```

Comment:
This recursive depth thing is interesting, but it could quickly blow up in terms of data volume. Space complexity here is something like `O(n!)` which is pretty scary :D.

The concept though of bringing in tracks from artists related to artists etc is pretty cool, but it'll need some contraints to be usable. E.g. max size of playlist, max number of songs per artists, etc

Reply:
yeah, i quickly realised this when i put in 2 and realised it totally crashed haha. i guess for now it's a placeholder until I have a better way of handling related artists' related artists' related tracks, etc. it'll be built upon at a later date :D